platform/ontology/api.go

```diff
@@ -62,80 +61,60 @@ func (p *Platform) CurrentBlockNumber() (int64, error) {
 		return 0, err
 	}
 	var height int64
-	if block.Error != 0 {
-		err = errors.E("explorer error")
-	}
 	if len(block.Result) > 0 {
 		height = (int64)(block.Result[0].Height)
 	}
 	return height, nil
 }
 
-func (p *Platform) GetBlockByNumber(num int64) (*blockatlas.Block, error) {
+func (p *Platform) GetBlockByNumber(num int64) (block *blockatlas.Block, err error) {
 	response, err := p.client.GetBlockByNumber(num)
 	if err != nil {
-		logger.Error("GetBlockByNumber", logger.Params{"platform": p.Coin().Symbol, "details": err.Error()})
 		return nil, err
 	}
-	var (
-		block blockatlas.Block
-		txs   []blockatlas.Tx
-	)
-	if response.Error == 0 {
-		block.ID = response.Result.Hash
-		block.Number = int64(response.Result.Height)
-		for _, txn := range response.Result.TxnList {
-			tx := new(blockatlas.Tx)
-			tx.ID = txn.TxnHash
-			tx.Block = uint64(txn.Height)
-			if txn.ConfirmFlag == 1 {
-				tx.Status = blockatlas.StatusCompleted
-			}
-			tx.Date = int64(txn.TxnTime)
-			tx.Coin = coin.Ontology().ID
-			txs = append(txs, *tx)
-		}
-		block.Txs = txs
-	}
-
-	return &block, nil
+	block = normalizeBlock(response.Result)
+	return
 }
 
-func Normalize(srcTx *Tx, assetName string) (tx blockatlas.Tx, ok bool) {
+func Normalize(srcTx Tx, assetName string) (tx *blockatlas.Tx, ok bool) {
 	if len(srcTx.TransferList) < 1 {
 		return tx, false
 	}
 	transfer := srcTx.TransferList[0]
 	fee := numbers.DecimalExp(srcTx.Fee, 9)
-	var status blockatlas.Status
-	if srcTx.ConfirmFlag == 1 {
-		status = blockatlas.StatusCompleted
-	} else {
-		status = blockatlas.StatusFailed
-	}
 
-	tx = blockatlas.Tx{
-		ID:     srcTx.TxnHash,
-		Coin:   coin.ONT,
-		Fee:    blockatlas.Amount(fee),
-		Date:   srcTx.TxnTime,
-		Block:  srcTx.Height,
-		Status: status,
-	}
+	tx = normalizeBlockTransaction(srcTx)
+	tx.Fee = blockatlas.Amount(fee)
 
 	switch assetName {
 	case ONTAssetName:
-		normalizeONT(&tx, &transfer)
+		normalizeONT(tx, transfer)
 	case ONGAssetName:
-		normalizeONG(&tx, &transfer)
+		normalizeONG(tx, transfer)
 	default: // unsupported asset
 		return tx, false
 	}
 
 	return tx, true
 }
 
-func normalizeONT(tx *blockatlas.Tx, transfer *Transfer) {
+func normalizeBlock(ontBlock Block) *blockatlas.Block {
+	var (
+		block blockatlas.Block
+		txs   []blockatlas.Tx
+	)
+	block.ID = ontBlock.Hash
+	block.Number = int64(ontBlock.Height)
+	for _, txn := range ontBlock.TxnList {
+		tx := normalizeBlockTransaction(txn)
+		tx.Meta = blockatlas.Transfer{}
```

Comment:
This does not make sense to me. Your transaction won't be complete? it does not have `from`, `to` and meta data for the transfer itself?